File: 20221102/모범답안/20221102_04/JoinToString_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.easy.JoinToString.JoinToString import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        arr = [1, 3, 0, 7, 9]
        prefix = "<"
        separator = ", "
        postfix = ">"

        # when
        result = solution(arr, prefix, separator, postfix)

        # then
        self.assertEqual("<1, 3, 0, 7, 9>", result)

    def test_2(self):
        # given
        arr = [1]
        prefix = "<"
        separator = ", "
        postfix = ">"

        # when
        result = solution(arr, prefix, separator, postfix)

        # then
        self.assertEqual("<1>", result)

    def test_3(self):
        # given
        arr = [1, 3, 0, 7, 9]
        prefix = ">"
        separator = ", "
        postfix = "<"

        # when
        result = solution(arr, prefix, separator, postfix)

        # then
        self.assertEqual(">1, 3, 0, 7, 9<", result)

    def test_4(self):
        # given
        arr = [12, 6, 6, 1, 1]
        prefix = "("
        separator = ", "
        postfix = ")"

        # when
        result = solution(arr, prefix, separator, postfix)

        # then
        self.assertEqual("(12, 6, 6, 1, 1)", result)

    def test_5(self):
        # given
        arr = [19, 0, 13, 16, 1]
        prefix = "["
        separator = ";"
        postfix = "]"

        # when
        result = solution(arr, prefix, separator, postfix)

        # then
        self.assertEqual("[19;0;13;16;1]", result)

    def test_6(self):
        # given
        arr = [3, 18, 17, 6, 9]
        prefix = ""
        separator = ""
        postfix = ""

        # when
        result = solution(arr, prefix, separator, postfix)

        # then
        self.assertEqual("3181769", result)


if __name__ == '__main__':
    unittest.main()

File: 20221102/모범답안/20221102_04/JoinToString_test.js
import assert from 'assert'
import solution from './JoinToString.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const arr = [1, 3, 0, 7, 9]
    const prefix = "<"
    const separator = ", "
    const postfix = ">"

    // when
    const result = solution(arr, prefix, separator, postfix)

    // then
    assert.deepStrictEqual("<1, 3, 0, 7, 9>", result)
    done();
  });

  it('test 2', function (done) {
    // given
    const arr = [1]
    const prefix = "<"
    const separator = ", "
    const postfix = ">"

    // when
    const result = solution(arr, prefix, separator, postfix)

    // then
    assert.deepStrictEqual("<1>", result)
    done();
  });

  it('test 3', function (done) {
    // given
    const arr = [1, 3, 0, 7, 9]
    const prefix = ">"
    const separator = ", "
    const postfix = "<"

    // when
    const result = solution(arr, prefix, separator, postfix)

    // then
    assert.deepStrictEqual(">1, 3, 0, 7, 9<", result)
    done();
  });

  it('test 4', function (done) {
    // given
    const arr = [12, 6, 6, 1, 1]
    const prefix = "("
    const separator = ", "
    const postfix = ")"

    // when
    const result = solution(arr, prefix, separator, postfix)

    // then
    assert.deepStrictEqual("(12, 6, 6, 1, 1)", result)
    done();
  });

  it('test 5', function (done) {
    // given
    const arr = [19, 0, 13, 16, 1]
    const prefix = "["
    const separator = ";"
    const postfix = "]"

    // when
    const result = solution(arr, prefix, separator, postfix)

    // then
    assert.deepStrictEqual("[19;0;13;16;1]", result)
    done();
  });

  it('test 6', function (done) {
    // given
    const arr = [3, 18, 17, 6, 9]
    const prefix = ""
    const separator = ""
    const postfix = ""

    // when
    const result = solution(arr, prefix, separator, postfix)

    // then
    assert.deepStrictEqual("3181769", result)
    done();
  });
})
